SOURCES/etc/xapi.d/plugins/ipmitool.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import json
+import sys
+import XenAPIPlugin
+
+sys.path.append(".")
+from xcpngutils import (
+    configure_logging,
+    run_command,
+    error_wrapped,
+    ProcessException,
+)
+
+
+@error_wrapped
+def run_ipmitool(args):
+    try:
+        output = run_command(["ipmitool", "sdr", "list"])
+    except ProcessException as e:
+        # Check if it is a "normal" error.
+        if "Could not open device" in e.stderr:
+            return (
+                False,
+                "IPMI device not found. Ensure the IPMI module is loaded and that your system supports IPMI.",
+            )
+        else:
+            raise e
+    else:
+        return True, output
```

Comment:
Maybe return directly the result without the `else`. Can be problematic on some linters/checkers.